### PR TITLE
feat: login?!

### PR DESCRIPTION
**Pull request by uday-jangra**

### CODE REVIEW

File: server/middleware/auth.middleware.js [hunk 1]
```diff
@@ -1,8 +1,27 @@
...
+        return res.sendStatus(403)
+      }
+      req.user = user
+      next()
+    } else {
+      res.sendStatus(403)
+    }
+  } catch (error) {
+    res.sendStatus(403)
+  }
```
Comment (georgevarghese185) [inline server/middleware/auth.middleware.js:26]: Reduce if nesting

File: server/repositories/user.repository.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import { db } from './mongo.js'
+import { ObjectId } from 'mongodb'
+
+const userCollection = db.collection('users')
+userCollection.createIndex('email')
+
+export async function getUser(email) {
```
Comment (georgevarghese185) [inline server/repositories/user.repository.js:7]: getUserByEmail

File: server/router/user.router.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      }
+
+      const isMatch = await bcrypt.compare(password, user.password)
+      if (!isMatch) {
+        return resp.status(400).json({ message: 'Invalid credentials' })
+      }
+
+      const token = jwt.sign({ userId: user._id }, jwtSecret, {
+        expiresIn: '1h',
+      })
```
Comment (georgevarghese185) [inline server/router/user.router.js:26]: Move this code to user service

File: server/router/user.router.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      return resp.status(400).json({ message: 'User already exists' })
+    }
+
+    const hashedPassword = await bcrypt.hash(password, 10)
+
+    await userService.registerUser({
+      name,
+      email,
+      hashedPassword,
+    })
```
Comment (georgevarghese185) [inline server/router/user.router.js:56]: move to user service

File: web/css/modal.css [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+.modal-body::-webkit-scrollbar-thumb {
+  background-color: #ccc;
+  border-radius: 4px;
+}
+
+/* Disable body scrolling when modal is open */
+body.modal-open {
+  overflow: hidden;
+}
```
Comment (georgevarghese185) [inline web/css/modal.css:112]: if you have the time see if this can be bootstrapped. Otherwise idc

File: web/services/user.service.js [hunk 6]
```diff
@@ -1,4 +1,5 @@
 class UserService {
+  url = 'http://localhost:8080/api'
```
Comment (georgevarghese185) [inline web/services/user.service.js:2]: add a API_URL variable to `.env.local` with this value

**Review by georgevarghese185 — APPROVED** [review]

File: server/repositories/mongo.js [hunk 7]
```diff
@@ -18,7 +18,7 @@ process.on('uncaughtException', (error) => {
     console.error(error.message)
     return
   }
-
+  console.log(error)
```
Comment (Osama671) [inline server/repositories/mongo.js:21]: what this

File: server/router/user.router.js [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+      resp.status(500).json({ message: 'Server error' })
+    }
+  } catch (err) {
+    resp.status(500).json({ error: err.message })
+  }
+})
+
+router.post('/register', async (req, resp) => {
+  const { name, email, password } = req.body
+  console.log(req.body)
```
Comment (Osama671) [inline server/router/user.router.js:33]: what this

File: web/context/AuthContext.jsx [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+      }
+    }
+    fetchUser()
+  }, [])
+
+  const login = async (email, password) => {
+    try {
+      setLoading(true)
+      const data = await userService.login(email, password)
+      console.log(data)
```
Comment (Osama671) [inline web/context/AuthContext.jsx:44]: Please eradicate, these console.logs are suffocating my memory sticks

**Pull request merged**